CaloReco/src/CaloHitMakerFast_module.cc

```diff
@@ -108,23 +108,25 @@ namespace mu2e {
        for (const auto& caloDigi : caloDigis)
        {
            int crystalID   = cal.caloIDMapper().crystalIDFromSiPMID(caloDigi.SiPMID());
-
-           double baseline = (caloDigi.waveform().at(0)+caloDigi.waveform().at(1)+caloDigi.waveform().at(2)+caloDigi.waveform().at(3))/4.0; 
-           double eDep     = (caloDigi.waveform().at(caloDigi.peakpos())-baseline)*calorimeterCalibrations->ADC2MeV(caloDigi.SiPMID()); 
+	   size_t nSamPed  = caloDigi.peakpos() > 3 ? 4 : std::max(caloDigi.peakpos()-1, 1);
+           double baseline(0);
+	   for (size_t i=0; i<nSamPed; ++i){ baseline += caloDigi.waveform().at(i);}
+	   baseline /= nSamPed;
```

Comment:
Again for future reference.  I like that you are using .at(i) instead of [i] for development.  For time critical code we should switch it back to [i] for production.  I know of several ways to way to write code so that we can flip between the two with a compiler switch.  When you are ready let's talk about it.

Reply:
Interesting. If this is something that affects several parts of the Offline it might be good to identify someone who has time to make this change.

Reply:
Do you have any summer students who could do it?

Reply:
No. The summer students working with my group have other projects assigned on the trigger front.

Reply:
I created an issue to followup on this general. 

If we do this in a few places the time cost is small.  If it's ubiquitous then it can add up to a large time cost that is invisible to profilers because it's distributed throughout the code.